paasta_tools/check_oom_events.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env python
+# Copyright 2015-2016 Yelp Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import argparse
+import json
+import sys
+import time
+from random import choice
+
+from pysensu_yelp import Status
+
+from paasta_tools import monitoring_tools
+from paasta_tools.chronos_tools import compose_check_name_for_service_instance
+from paasta_tools.cli.utils import get_instance_config
+from paasta_tools.utils import DEFAULT_SOA_DIR
+from paasta_tools.utils import get_services_for_cluster
+from paasta_tools.utils import load_system_paasta_config
+
+try:
+    from scribereader import scribereader
+except ImportError:
+    scribereader = None
+
+
+OOM_EVENTS_STREAM = 'tmp_paasta_oom_events'
+
+
+def parse_args(args):
+    parser = argparse.ArgumentParser(description=(
+        'Check the %s stream and report to Sensu if'
+        ' there are any OOM events.' % OOM_EVENTS_STREAM
+    ))
+    parser.add_argument(
+        '-d', '--soa-dir', dest="soa_dir", default=DEFAULT_SOA_DIR,
+        help="define a different soa config directory",
+    )
+    parser.add_argument(
+        '-r', '--realert-every', dest="realert_every", type=int, default=1,
+        help="Sensu 'realert_every' to use.",
+    )
+    parser.add_argument(
+        '-s', '--superregion', dest="superregion", required=True,
+        help="The superregion to read OOM events from.",
+    )
+    return parser.parse_args(args)
+
+
+def oom_events(cluster, superregion, num_lines=1000):
+    """Read the latest 'num_lines' lines from OOM_EVENTS_STREAM and iterate over them."""
+    host_port = choice(scribereader.get_default_scribe_hosts(tail=True))
+    stream = scribereader.get_stream_tailer(
+        OOM_EVENTS_STREAM, host_port['host'], host_port['port'], True,
+        num_lines, superregion=superregion,
+    )
+    for line in stream:
+        try:
+            j = json.loads(line)
+            if j.get('cluster', '') == cluster:
+                yield j
+        except json.decoder.JSONDecodeError:
+            pass
+
+
+def latest_oom_events(cluster, superregion, interval=60):
+    """
+    :returns: {(service, instance): [(hostname, container_id, process_name), ...] }
```

Comment:
These complicated return thingys are easy to get wrong. Can you implement this as a named tuple instead?